guardian/prosemirror-typerighter
# feat: Provide tooltips explaining meaning of each header

<blockquote>"At the moment, in our new Typerighter UI, we have a single word to describe each category. There's still plenty of room for ambiguity here!"</blockquote>

Users have found the heading titles to not be as explanatory as they could be of the purpose of the checks in prosemirror-typerighter. They may wonder what the distinction between 'Amend' and 'Review' is, or why 'OK' matches are being highlighted.

This PR adds tooltips to the match headers, where we can explain what they mean in more detail.

![2022-08-02 15 44 22](https://user-images.githubusercontent.com/34686302/182403121-e6fe7c37-fa54-4473-a75c-4487c7c287b4.gif)

I've used some provisional wording, but it would be good to get some consensus on what useful wording would be for each category. It would be good to have some feedback on the current wording.

This is:
| Match type | Wording |
| --- | --- |
| Amend | "Typerighter doesn’t think these matches are correct." |
| Review | "Typerighter can’t figure out whether these matches are correct or not. They might be worth reviewing." |
| OK | "Typerighter thinks these matches are correct." |

## How to test
1. Run `prosemirror-typerighter` and `typerighter` locally, according to the instructions in their READMEs. You'll also need credentials to run them locally - these can be shared by running another project with authentication locally, for example `flexible-content`.
2. Open this application, and click 'Check'. There should be a tooltip in each match type header (i.e. Amend, Review, OK) - and hovering over this should show a tooltip with relevant wording.
    - Does the styling make sense in the context of Composer, alongside its other tooltips and general design?
    - Does the wording of the descriptions make sense?
